infra-provider-kafka/src/main/java/com/homeaway/streamingplatform/provider/impl/KafkaInfraManager.java

```diff
@@ -36,56 +37,44 @@
 
 
 /**
- * The type Stream infra manager.
+ * An {@link InfraManager} implementation backed by a Kafka Streams {@link GlobalKTable}.
  */
 @SuppressWarnings({"unused", "WeakerAccess"})
 @Slf4j
 public class KafkaInfraManager implements InfraManager {
 
-    private KafkaStreams infraKStreams;
-    private String infraStateStoreName;
-    private ReadOnlyKeyValueStore<ClusterKey, ClusterValue> store;
-    /**
-     * The constant INFRAMANAGER_TOPIC.
-     */
     public static final String INFRAMANAGER_TOPIC = "infraManagerTopic";
-    /**
-     * The constant INFRAMANAGER_STATE_STORE.
-     */
     public static final String INFRAMANAGER_STATE_STORE = "infraManagerStateStoreName";
-    /**
-     * The constant INFRA_KSTREAM_PROPS.
-     */
     public static final String INFRA_KSTREAM_PROPS = "infraKStreamsProperties";
 
+    private String infraStateStoreName;
+    private KafkaStreams infraKStreams;
+    private GlobalKTable<ClusterKey, ClusterValue> kTable;
+    private ReadOnlyKeyValueStore<ClusterKey, ClusterValue> store;
+
     @SuppressWarnings("unchecked")
     @Override
     public void configure(Map<String, Object> configs) {
-        KStreamBuilder infraKStreamBuilder = new KStreamBuilder();
-        String infraManagerTopic;
-        Properties infraKStreamsProperties = new Properties();
-
         // Get the infra manager topic name
         Validate.validState(configs.containsKey(INFRAMANAGER_TOPIC), "Infra Manager Topic name is not provided.");
-        infraManagerTopic = configs.get(INFRAMANAGER_TOPIC).toString();
-
+        String infraManagerTopic = configs.get(INFRAMANAGER_TOPIC).toString();
         log.info("Infra Manager Topic Name Read: {}", infraManagerTopic);
 
         // Get the infra state store name
         Validate.validState(configs.containsKey(INFRAMANAGER_STATE_STORE), "Infra Manager State Store name is not provided.");
         infraStateStoreName = configs.get(INFRAMANAGER_STATE_STORE).toString();
-
         log.info("Infra Manager State Store Name Read: {}", infraStateStoreName);
 
         // Populate our kstreams properties map
+        Properties infraKStreamsProperties = new Properties();
         Validate.validState(configs.containsKey(INFRA_KSTREAM_PROPS), "InfraKStreams properties is not provided.");
         Map<String, Object> infraKStreamsPropertiesMap = (Map<String, Object>) configs.get(INFRA_KSTREAM_PROPS);
         infraKStreamsPropertiesMap.forEach(infraKStreamsProperties::put);
-
         log.info("Infra KStreams Properties: {}", infraKStreamsProperties);
 
         // initialize the kstreams processor
-        infraKStreamBuilder.globalTable(infraManagerTopic, infraStateStoreName);
+        KStreamBuilder infraKStreamBuilder = new KStreamBuilder();
+        kTable = infraKStreamBuilder.globalTable(infraManagerTopic, infraStateStoreName);
```

Comment:
Does this require proper closing during graceful shutdown?
Do any of these require proper closing during graceful shutdown?

If yes, add a comment and open an issue (lets not block this PR)

Reply:
Well, there is no close method on `GlobalKTable` interface (doesn't implement `Closeable` either). 

It seems `KafkaStreams.close()` shutsdown all revelant threads and objects.